src/app/data_harvester/processes/macos_freebsd.rs

```diff
@@ -50,11 +49,11 @@ where
         };
 
         let pcu = {
-            let p = process_val.cpu_usage() as f64 / num_processors;
-            if p.is_nan() {
-                process_val.cpu_usage() as f64
-            } else {
+            let p = process_val.cpu_usage() as f64;
+            if per_core_percentage || p.is_nan() {
                 p
+            } else {
+                p / (sys.cpus().len() as f64)
```

Comment:
Same as my other comment, the `p.is_nan()` check is mostly there to check the result of the division.

Reply:
Same as above.